From 5b8c53a6bb11c8aa9b575b5a8155c25c5085f349 Mon Sep 17 00:00:00 2001
From: Max Krummenacher <max.oss.09@gmail.com>
Date: Fri, 6 Nov 2015 12:22:35 +0100
Subject: [PATCH] bignum.cc: disable warning from gcc 5

addresses:
  ../../third_party/WebKit/Source/wtf/dtoa/bignum.cc:105:10: error: assuming
  signed overflow does not occur when assuming that (X + c) < X is always
  false [-Werror=strict-overflow]
      void Bignum::AssignDecimalString(Vector<const char> value) {
           ^

Signed-off-by: Max Krummenacher <max.oss.09@gmail.com>

Upstream-Status: Pending
See the discussion on the issue in Chromium upstream:
https://groups.google.com/a/chromium.org/forum/#!topic/chromium-dev/3uwBwunIa7g
---
 third_party/WebKit/Source/wtf/dtoa/bignum.cc | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/third_party/WebKit/Source/wtf/dtoa/bignum.cc b/third_party/WebKit/Source/wtf/dtoa/bignum.cc
index a000b46..6c6d336 100644
--- a/third_party/WebKit/Source/wtf/dtoa/bignum.cc
+++ b/third_party/WebKit/Source/wtf/dtoa/bignum.cc
@@ -102,7 +102,10 @@
     }
 
 
+#pragma GCC diagnostic push
+#pragma GCC diagnostic warning "-Wstrict-overflow"
     void Bignum::AssignDecimalString(Vector<const char> value) {
+#pragma GCC diagnostic pop
         // 2^64 = 18446744073709551616 > 10^19
         const int kMaxUint64DecimalDigits = 19;
         Zero();
@@ -699,7 +699,10 @@
     }


+#pragma GCC diagnostic push
+#pragma GCC diagnostic warning "-Wstrict-overflow"
     void Bignum::Align(const Bignum& other) {
+#pragma GCC diagnostic pop
         if (exponent_ > other.exponent_) {
             // If "X" represents a "hidden" digit (by the exponent) then we are in the
             // following case (a == this, b == other):
@@ -715,7 +715,10 @@
             for (int i = used_digits_ - 1; i >= 0; --i) {
                 bigits_[i + zero_digits] = bigits_[i];
             }
+#pragma GCC diagnostic push
+#pragma GCC diagnostic warning "-Wstrict-overflow"
             for (int i = 0; i < zero_digits; ++i) {
+#pragma GCC diagnostic pop
                 bigits_[i] = 0;
             }
             used_digits_ += zero_digits;
-- 
1.8.4.5
